Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -397,7 +397,7 @@ protected override void Dispose(bool disposing)
                 }
                 else
                 {
-                    SfSession?.close();
+                    SfSession?.closeNonBlocking();

```

Comment:
When the dispose method is called from the Garbage collector is non-deterministic so it could be an case when the SfSession is already processed by the garbage collector and generate unexpected behaviors. So instead of calling the closeNonBlocking method here, we should implement IDisposable interface in the SFSession class and handle the behavior there.

Disposable documentation https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#the-disposebool-method-overload

Reply:
we discussed it to leave as is for now.